### PR TITLE
refactor(store): use sqlite rather than memory store

### LINKED ISSUE #201: Feat: add option to use the SQLite store with the Astarte Sdk
Add an option in the config to be able to use the SQLite store with the Astarte SDK.

The database would be store in the already present store_directory, we should add a boolean option to the [`MessageHubOptions`](https://github.com/astarte-platform/astarte-message-hub/blob/e673860701ca42a2e663d9c1be279669b1b872fc/src/config/mod.rs#L52C12-L52C29) to enable the SQLite store.

### PR DESCRIPTION
**Pull request by rgallor**
Close #201

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -52,8 +52,10 @@ struct Cli {
 }
 
 #[tokio::main]
-async fn main() -> Result<(), AstarteMessageHubError> {
-    env_logger::init();
+async fn main() -> color_eyre::Result<()> {
+    color_eyre::install()?;
```
[the remark this answers is not in the document]
Comment (rgallor) [reply src/main.rs:56]: It avoids creating new error variants in the binary and it reports the full error backtrace from `thiserror`. It's a fork of anyhow

[the remark this answers is not in the document]
Comment (harlem88) [reply src/main.rs:56]: So we have to add two crates to avoid creating new error variant?

[the remark this answers is not in the document]
Comment (rgallor) [reply src/main.rs:56]: It also adds new functionalities that can be used in the future, but I get your point and it is not "necessary" to use `eyre`.

**Comment by rgallor** [comment]
As discussed internally, I maintained the `eyre` crate and changed `color_eyre` with `stable_eyre`

**Review by joshuachp — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**